app/src/main/java/com/github/libretube/ui/models/PlaylistViewModel.kt

```diff
@@ -1,7 +1,109 @@
 package com.github.libretube.ui.models
 
+import android.os.Parcelable
+import androidx.annotation.StringRes
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.createSavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.github.libretube.R
+import com.github.libretube.api.PlaylistsHelper
+import com.github.libretube.api.obj.Playlists
+import com.github.libretube.api.obj.StreamItem
+import com.github.libretube.constants.IntentData
+import com.github.libretube.util.PlayingQueue
+import kotlinx.coroutines.launch
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.RawValue
 
-class PlaylistViewModel : ViewModel() {
-    var lastSelectedPlaylistId: String? = null
+class PlaylistViewModel(
+    private val savedStateHandle: SavedStateHandle,
+) : ViewModel() {
+
+    private val _uiState = savedStateHandle.getStateFlow(UI_STATE, UiState())
+    val uiState = _uiState.asLiveData()
+
+    init {
+        fetchPlaylists()
+    }
+
+    fun fetchPlaylists() {
+        viewModelScope.launch {
+            kotlin.runCatching {
+                PlaylistsHelper.getPlaylists()
+            }.onSuccess { playlists ->
+                savedStateHandle[UI_STATE] = _uiState.value.copy(
+                    playlists = playlists.filterNot { list -> list.name.isNullOrEmpty() }
+                )
+            }.onFailure {
+                savedStateHandle[UI_STATE] = _uiState.value.copy(
+                    message = UiState.Message(R.string.unknown_error)
+                )
+            }
+        }
+    }
+
+    fun onPostiveButtonClick(playlistIndex: Int) {
```

Comment:
nit: this should probably be called something like `addToPlaylistAtIndex` instead, `onPositiveButtonClick` is not that descriptive.